nemo_aligner/data/nlp/datasets.py

```diff
@@ -403,10 +432,153 @@ def __getitem__(self, idx):
             "rejected_labels": labels_reject_tokens,
             "chosen_reward": payload.get("chosen_reward", self.default_chosen_reward),
             "rejected_reward": payload.get("rejected_reward", self.default_rejected_reward),
+            "ignore_example": ignore_example,
         }
         return output
 
 
+class DPOPackedDataset(DPOModelDataset):
+    """A dataset class for DPO with sequence packing. Data is expected to be 
+    pre-tokenized and pre-packed using examples/nlp/data/dpo/prepare_packed_dpo_dataset.py.
+    """
+
+    def __init__(
+        self,
+        cfg,
+        tokenizer,
+        name,
+        data_prefix,
+        documents,
+        data,
+        seq_length,
+        seed,
+        drop_last=True,  # return_cu_seqlen: bool = True ## should always be true
+    ):
+
+        super().__init__(cfg, tokenizer, name, data_prefix, documents, data, seq_length, seed, drop_last)
+        self.data_prefix = data_prefix
+
+    def __getitem__(self, idx):
+        return self.data[idx]
+
+    def __len__(self):
```

Comment:
do some of these need to be defined if inheriting from DPOModelDataset?

Reply:
`__len__` was redundant. I remove it.